### ISSUE TITLE
Knitr lessons

### ISSUE THREAD

**Issue by sritchie73** [swcarpentry/r-novice-gapminder#11]
It would be really nice to have a small lesson on the use of knitr for report generation within RStudio. @nfaux ran a workshop last week with a knitr lesson, but there are not yet any corresponding lesson materials. We've got his project file to use as reference materials.

**Comment by naupaka** [comment]
@kbroman and @karawoo developed some knitr lessons for a SWC workshop they taught WSU, perhaps these could be used as the starting point?

**Comment by karawoo** [comment]
The lesson is [here](https://github.com/karawoo/2015-04-27-wsu/blob/gh-pages/syllabus/Rmarkdown.Rmd); all the credit goes to @kbroman for this one. I like the idea of having a knitr lesson. People at our workshop thought this was really cool.

**Comment by hdashnow** [comment]
+1 for knitr lessons. Shall we copy in @kbroman's lesson and then hack from there?

**Comment by naupaka** [comment]
I think that's a great idea. Especially since the formal title of the lessons is "R for Reproducible Scientific Analysis". I think pulling in the material from @kbroman would be a great place to start. I'd sequence this lesson towards the end, since it's quite synthetic? 

Dear internets, PRs welcome :smile:

**Comment by tomwright01** [comment]
I have just taught a python workshop which introduced the `jupyter notebook`. To get the participants to actually use a notebook I tried to use a problem solving exercise where I provided a skeleton notebook which they then completed (in pairs / small groups). The aims for this were a) get participants to create a notebook, and b) get participants to use the documentation for specific functions. The sample notebook `ConeCounting.ipynb` as well as the completed version `CouneCounting-Complete.ipynb` can be seen here: https://github.com/tomwright01/imp-parallel
The approach worked really well with this group and pairing the participants helped dealing with the wide range of background knowledge.
